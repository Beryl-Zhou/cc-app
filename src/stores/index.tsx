import { enableStaticRendering } from "mobx-react-lite";
import React, { createContext, ReactNode, useContext } from "react";
import { RootStore, RootStoreHydration } from "./RootStore";

const IS_SSR = typeof window === "undefined";
enableStaticRendering(IS_SSR);

let store: RootStore;
const StoreContext = createContext<RootStore | undefined>(undefined);
// StoreContext.displayName = "StoreContext";

export function useStore() {
  const context = useContext(StoreContext);
  if (context === undefined) {
    throw new Error("useStore must be used within StoreProvider");
  }

  return context;
}

export function useCounterStore() {
  const { counterStore } = useStore();
  return counterStore;
}

export function useSizeSwitcherStore() {
  const { sizeSwitcherStore } = useStore();
  return sizeSwitcherStore;
}

export function StoreProvider({
  children,
  hydrationData,
}: {
  children: ReactNode;
  hydrationData?: RootStoreHydration;
}) {
  const store = initializeStore(hydrationData);

  return <StoreContext.Provider value={store}>{children}</StoreContext.Provider>;
}

function initializeStore(initialData?: RootStoreHydration): RootStore {
  const _store = store ?? new RootStore();

  if (initialData) {
    _store.hydrate(initialData);
  }
  // For SSG and SSR always create a new store
  if (IS_SSR) return _store;
  // Create the store once in the client
  if (!store) store = _store;

  return _store;
}
