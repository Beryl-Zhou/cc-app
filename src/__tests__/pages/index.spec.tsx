import { render } from "@testing-library/react";
import Home from "src/pages";
import { StoreProvider } from '../../stores'

describe("Home", () => {
  it("renders homepage", () => {
    const { container } = render(<StoreProvider><Home /></StoreProvider>);
    expect(container).toMatchSnapshot();
  });
});
