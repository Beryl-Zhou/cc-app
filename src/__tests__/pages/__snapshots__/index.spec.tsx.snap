// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders homepage 1`] = `
<div>
  <div>
    <main>
      <h1
        class="sc-bczRLJ dgihId"
      >
        Hello world!
      </h1>
      <div
        class="buttons-wrap"
      >
        <button
          class="chakra-button css-5c2176"
          type="button"
        >
          Big
        </button>
        <button
          class="chakra-button css-5c2176"
          disabled=""
          type="button"
        >
          Small
        </button>
      </div>
      <p
        class="chakra-text css-1ezsviu"
      >
        0
      </p>
      <div
        class="buttons-wrap"
      >
        <button
          class="chakra-button css-5c2176"
          disabled=""
          type="button"
        >
          PAUSE
        </button>
        <button
          class="chakra-button css-5c2176"
          type="button"
        >
          START
        </button>
      </div>
      <div
        class="links-wrap"
      >
        <a
          href="/?start=500"
        >
          start from 
          <strong>
            500
          </strong>
        </a>
        <a
          href="/?start=1000"
        >
          start from 
          <strong>
            1000
          </strong>
        </a>
      </div>
    </main>
    <footer>
      this is footer
    </footer>
  </div>
</div>
`;
